metrics/ci_integrations/test/client/github_actions/test_utils/mock/github_actions_mock_server.dart

```diff
@@ -0,0 +1,233 @@
+          ),
+          dispatcher: _notFoundResponse,
+        ),
+        RequestHandler.get(
+          pathMatcher: ExactPathMatcher(
+            '/repos/owner/name/actions/runs/1/artifacts',
+          ),
+          dispatcher: _workflowRunArtifactsResponse,
+        ),
+        RequestHandler.get(
+          pathMatcher: ExactPathMatcher(
+            '/repos/owner/name/actions/runs/test/artifacts',
+          ),
+          dispatcher: _notFoundResponse,
+        ),
+        RequestHandler.get(
+          pathMatcher: ExactPathMatcher(
+            '/repos/owner/name/actions/runs/1/timing',
+          ),
+          dispatcher: _workflowUsageResponse,
+        ),
+        RequestHandler.get(
+          pathMatcher: ExactPathMatcher(
+            '/repos/owner/name/actions/runs/test/timing',
+          ),
+          dispatcher: _notFoundResponse,
+        ),
+        RequestHandler.get(
+          pathMatcher: ExactPathMatcher(
+            '/repos/owner/name/actions/artifacts/artifact_id/zip',
+          ),
+          dispatcher: _downloadArtifactResponse,
+        ),
+        RequestHandler.get(
+          pathMatcher: ExactPathMatcher(
+            '/artifact/download',
+          ),
+          dispatcher: _downloadResponse,
+        ),
+        RequestHandler.get(
+          pathMatcher: ExactPathMatcher(
+            '/repos/owner/name/actions/artifacts/test/zip',
+          ),
+          dispatcher: _notFoundResponse,
+        ),
+      ];
+
+  /// Responses with a list of all workflow runs for a specific workflow.
+  Future<void> _workflowRunsResponse(HttpRequest request) async {
+    final status = _extractStatus(request);
+
+    List<WorkflowRun> workflowRuns = _generateWorkflowRuns(status);
+
+    workflowRuns = _paginate(
+      workflowRuns,
+      _extractPerPage(request),
+      _extractPage(request),
+    );
+
+    final _response = {
+      'total_count': workflowRuns.length,
+      'workflow_runs': workflowRuns.map((run) => run.toJson()).toList(),
+    };
+
+    request.response.write(jsonEncode(_response));
+
+    await request.response.flush();
+    await request.response.close();
+  }
+
+  /// Responses with a list of artifacts for a specific workflow run.
+  Future<void> _workflowRunArtifactsResponse(HttpRequest request) async {
+    List<WorkflowRunArtifact> artifacts = _generateArtifacts();
+
+    artifacts = _paginate(
+      artifacts,
+      _extractPerPage(request),
+      _extractPage(request),
+    );
+
+    final _response = {
+      'total_count': artifacts.length,
+      'artifacts': artifacts.map((artifact) => artifact.toJson()).toList(),
+    };
+
+    request.response.write(jsonEncode(_response));
+
+    await request.response.flush();
+    await request.response.close();
+  }
+
+  /// Responses with the number of billable minutes and total run time
+  /// for a specific workflow run.
+  Future<void> _workflowUsageResponse(HttpRequest request) async {
+    final _response = {"run_duration_ms": 500000};
```

Comment:
Would it be more readable to use the dart objects (`WorkflowRunDuration` in this situation) for mock data?